Added ROC Indicator

abhishekmittal15 opened the pull request:
Fixed #18 

Added a file `priceRateOfChange.ts` which houses the `roc` indicator function. 
Also added a test file `priceRateOfChange.test.ts` which runs successfully

@cinar , Looking forward for feedback. 

Also I wanted to ask you about input validation and exception handling. For example, in this indicator if the input array has a $0$ then, the code will crash due to division by 0, so should there be any warnings to be raised when the input is passed initially to the function

cinar commented:
Thank you very much for the PR Abhishek!  Great point, I typically try to document those in the index.md file here in the same folder. Definitely worth calling that out there.

Two minor things before merging your changes:
- If you can take a moment to add a section to the documentation in index.md file in the same folder, and also please link to it from the README.md file also.
- Also let's put the content of the else statement inside curly brackets as well.

Many thanks!

abhishekmittal15 commented:
Hey, 
I have updated the docs and made the code changes you suggested. 
I just noticed that I had interpreted the indicator formula wrong, I was thinking of this indicator as taking the relative difference between the starting and ending values of a window of size `period`, but this is not the case, I should have taken my close[i-period] as the first value to the left of this window rather than the first value of the window. So I have fixed the implementation and tests accordingly. Sorry for the confusion.
